core/templates/pages/exploration-player-page/layout-directives/learner-view-info.component.spec.ts

```diff
@@ -0,0 +1,194 @@
+  let statsReportingService: StatsReportingService;
+  let urlInterpolationService: UrlInterpolationService;
+  let urlService: UrlService;
+  let storyViewerBackendApiService: StoryViewerBackendApiService;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ],
+      declarations: [
+        LearnerViewInfoComponent,
+        MockTranslatePipe
+      ],
+      providers: [
+        NgbModal,
+        ContextService,
+        LearnerViewInfoBackendApiService,
+        LoggerService,
+        ReadOnlyExplorationBackendApiService,
+        SiteAnalyticsService,
+        StatsReportingService,
+        UrlInterpolationService,
+        UrlService,
+        StoryViewerBackendApiService
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(LearnerViewInfoComponent);
+    componentInstance = fixture.componentInstance;
+    contextService = TestBed.inject(ContextService);
+    learnerViewInfoBackendApiService = TestBed.inject(
+      LearnerViewInfoBackendApiService);
+    loggerService = TestBed.inject(LoggerService);
+    readOnlyExplorationBackendApiService = TestBed.inject(
+      ReadOnlyExplorationBackendApiService);
+    siteAnalyticsService = TestBed.inject(SiteAnalyticsService);
+    statsReportingService = TestBed.inject(StatsReportingService);
+    urlInterpolationService = TestBed.inject(UrlInterpolationService);
+    urlService = TestBed.inject(UrlService);
+    storyViewerBackendApiService = TestBed.inject(StoryViewerBackendApiService);
+  });
+
+  afterEach(() => {
+    componentInstance.ngOnDestory();
+  });
+
+  it('should initialize', fakeAsync(() => {
+    let explorationId = 'expId';
+    let explorationTitle = 'Exploration Title';
+    let topicUrl = 'topic_url';
+
+    spyOn(urlService, 'getPathname').and.returnValue('/explore/');
+    spyOn(contextService, 'getExplorationId').and.returnValue(explorationId);
+    spyOn(readOnlyExplorationBackendApiService, 'fetchExplorationAsync')
+      .and.returnValue(Promise.resolve({
+        exploration: {
+          title: explorationTitle
+        }
+      } as FetchExplorationBackendResponse));
+
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    spyOn(componentInstance, 'getTopicUrl').and.returnValue(topicUrl);
+    spyOn(urlService, 'getTopicUrlFragmentFromLearnerUrl').and.returnValue('');
+    spyOn(urlService, 'getClassroomUrlFragmentFromLearnerUrl')
+      .and.returnValue('');
+    spyOn(urlService, 'getStoryUrlFragmentFromLearnerUrl').and.returnValue('');
+    spyOn(storyViewerBackendApiService, 'fetchStoryDataAsync').and.returnValue(
+      Promise.resolve(new StoryPlaythrough('', [], '', '', '', '')));
+    spyOn(statsReportingService, 'setTopicName');
+    spyOn(siteAnalyticsService, 'registerCuratedLessonStarted');
+
+    componentInstance.ngOnInit();
+    tick();
+    tick();
```

Comment:
Is 2 `tick()` function calls required here?

Reply:
Yes, there are two async backend API calls in ngOnInit hook.